src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -81,42 +81,46 @@ public ITest Build(Assembly assembly, IDictionary<string, object> options)
 #endif
 
             string assemblyPath = AssemblyHelper.GetAssemblyPath(assembly);
-            return Build(assembly, assemblyPath, options);
+            string assemblyName = assemblyPath.IndexOfAny(Path.GetInvalidPathChars()) == -1
+                ? Path.GetFileName(assemblyPath)
+                : AssemblyHelper.GetAssemblyName(assembly).FullName;
+
+            return Build(assembly, assemblyName, options);
         }
 
         /// <summary>
         /// Build a suite of tests given the filename of an assembly
         /// </summary>
-        /// <param name="assemblyName">The filename of the assembly from which tests are to be built</param>
+        /// <param name="assemblyNameOrPath">The filename or path of the assembly from which tests are to be built</param>
```

Comment:
I think this should actually say `the assembly name or filename` or `the name or location of the assembly`.